Typings issue in v3.3.0

When using Listr2 in a TypeScript project I am getting the following TSC error when using v3.3.0:
```
../../node_modules/listr2/dist/lib/task-wrapper.d.ts:27:5 - error TS2416: Property 'run' in type 'TaskWrapper<Ctx, Renderer>' is not assignable to the same property in base type 'ListrTaskWrapper<Ctx, Renderer>'.
  Type '(ctx: Ctx) => Promise<void>' is not assignable to type '(ctx?: Ctx | undefined, task?: ListrTaskWrapper<Ctx, Renderer> | undefined) => Promise<void>'.
    Types of parameters 'ctx' and 'ctx' are incompatible.
      Type 'Ctx | undefined' is not assignable to type 'Ctx'.
        'Ctx' could be instantiated with an arbitrary type which could be unrelated to 'Ctx | undefined'.
          Type 'undefined' is not assignable to type 'Ctx'.
            'Ctx' could be instantiated with an arbitrary type which could be unrelated to 'undefined'.

27     run(ctx: Ctx): Promise<void>;
```
It compiles fine with the previous v3.2.3.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,10 @@
+## [3.3.1-beta.1](https://github.com/cenk1cenk2/listr2/compare/v3.3.0...v3.3.1-beta.1) (2021-01-24)
+
+
+### Bug Fixes
+
+* rollback types ([7935241](https://github.com/cenk1cenk2/listr2/commit/7935241ef91ed36434242ee00ed838c1fe0979e2)), closes [#276](https://github.com/cenk1cenk2/listr2/issues/276)
+
 # [3.3.0](https://github.com/cenk1cenk2/listr2/compare/v3.2.3...v3.3.0) (2021-01-20)
 
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,10 @@
+## [3.3.1-beta.3](https://github.com/cenk1cenk2/listr2/compare/v3.3.1-beta.2...v3.3.1-beta.3) (2021-01-31)
+
+
+### Bug Fixes
+
+* try fix typings issue ([4dd9013](https://github.com/cenk1cenk2/listr2/commit/4dd901356dffa24d3a3d41cf829896da7890e17e))
+
 ## [3.3.1-beta.2](https://github.com/cenk1cenk2/listr2/compare/v3.3.1-beta.1...v3.3.1-beta.2) (2021-01-25)
 
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,10 @@
+## [3.3.1-beta.2](https://github.com/cenk1cenk2/listr2/compare/v3.3.1-beta.1...v3.3.1-beta.2) (2021-01-25)
+
+
+### Bug Fixes
+
+* update interface ([2cb2cd6](https://github.com/cenk1cenk2/listr2/commit/2cb2cd637efdbdadfff548ed0c9cccbb347ca5f6))
+
 ## [3.3.1-beta.1](https://github.com/cenk1cenk2/listr2/compare/v3.3.0...v3.3.1-beta.1) (2021-01-24)
 
 

--- a/src/interfaces/listr.interface.ts
+++ b/src/interfaces/listr.interface.ts
@@ -66,7 +66,7 @@ export interface ListrTaskObject<Ctx, Renderer extends ListrRendererFactory> ext
    */
   check: (ctx: Ctx) => void
   /** Run the current task. */
-  run: (ctx: Ctx, wrapper: ListrTaskWrapper<Ctx, Renderer>) => Promise<void>
+  run(ctx: Ctx, wrapper: ListrTaskWrapper<Ctx, Renderer>): Promise<void>
   /** Options for listr itself. */
   options: ListrOptions
   /** Options for the current renderer of the task. */
@@ -153,7 +153,7 @@ export interface ListrTaskWrapper<Ctx, Renderer extends ListrRendererFactory> {
   /** Skip current task. */
   skip: (message?: string) => void
   /** Run this task. */
-  run: (ctx?: Ctx, task?: ListrTaskWrapper<Ctx, Renderer>) => Promise<void>
+  run(ctx?: Ctx, task?: ListrTaskWrapper<Ctx, Renderer>): Promise<void>
   /**
    * Create a new Enquirer prompt using prompt options.
    *

--- a/src/interfaces/listr.interface.ts
+++ b/src/interfaces/listr.interface.ts
@@ -10,7 +10,7 @@ import { Listr } from '@root/index'
 import { PromptOptions } from '@utils/prompt.interface'
 
 /** Listr Default Context */
-export type ListrContext = any
+export type ListrContext = any | undefined
 
 /** The default renderer value used in Listr2 applications */
 export type ListrDefaultRendererValue = 'default'
@@ -64,9 +64,9 @@ export interface ListrTaskObject<Ctx, Renderer extends ListrRendererFactory> ext
   /**
    * A function to check whether this task should run at all via enable.
    */
-  check(ctx: Ctx): void
+  check: (ctx: Ctx) => void
   /** Run the current task. */
-  run(ctx: Ctx, wrapper: ListrTaskWrapper<Ctx, Renderer>): Promise<void>
+  run: (ctx: Ctx, wrapper: ListrTaskWrapper<Ctx, Renderer>) => Promise<void>
   /** Options for listr itself. */
   options: ListrOptions
   /** Options for the current renderer of the task. */
@@ -76,25 +76,25 @@ export interface ListrTaskObject<Ctx, Renderer extends ListrRendererFactory> ext
   /** A hook to refresh render if desired. */
   renderHook$: Subject<void>
   /** Returns whether this task has subtasks. */
-  hasSubtasks(): boolean
+  hasSubtasks: () => boolean
   /** Returns whether this task is in progress. */
-  isPending(): boolean
+  isPending: () => boolean
   /** Returns whether this task is skipped. */
-  isSkipped(): boolean
+  isSkipped: () => boolean
   /** Returns whether this task has been completed. */
-  isCompleted(): boolean
+  isCompleted: () => boolean
   /** Returns whether this task has an active rollback task going on. */
-  isRollingBack(): boolean
+  isRollingBack: () => boolean
   /** Returns whether the rollback action was successful. */
-  hasRolledBack(): boolean
+  hasRolledBack: () => boolean
   /** Returns whether enabled function resolves to true. */
-  isEnabled(): boolean
+  isEnabled: () => boolean
   /** Returns whether this task has a prompt inside. */
-  isPrompt(): boolean
+  isPrompt: () => boolean
   /** Returns whether this task has been failed. */
-  hasFailed(): boolean
+  hasFailed: () => boolean
   /** Returns whether this task actually has a title. */
-  hasTitle(): boolean
+  hasTitle: () => boolean
 }
 
 export interface ListrTask<Ctx = ListrContext, Renderer extends ListrRendererFactory = any> {

--- a/src/interfaces/listr.interface.ts
+++ b/src/interfaces/listr.interface.ts
@@ -39,7 +39,7 @@ export interface ListrTaskObject<Ctx, Renderer extends ListrRendererFactory> ext
   /** Output data from the task. */
   output?: string
   /** The task object itself, to further utilize it. */
-  task: (ctx: Ctx, task: ListrTaskWrapper<Ctx, Renderer>) => void | ListrTaskResult<Ctx>
+  task(ctx: Ctx, task: ListrTaskWrapper<Ctx, Renderer>): void | ListrTaskResult<Ctx>
   /** Skip current task. */
   skip: boolean | string | ((ctx: Ctx) => boolean | string | Promise<boolean> | Promise<string>)
   /** Extend current task with multiple subtasks. */
@@ -64,9 +64,9 @@ export interface ListrTaskObject<Ctx, Renderer extends ListrRendererFactory> ext
   /**
    * A function to check whether this task should run at all via enable.
    */
-  check: (ctx: Ctx) => void
+  check(ctx: Ctx): void
   /** Run the current task. */
-  run: (ctx: Ctx, wrapper: ListrTaskWrapper<Ctx, Renderer>) => Promise<void>
+  run(ctx: Ctx, wrapper: ListrTaskWrapper<Ctx, Renderer>): Promise<void>
   /** Options for listr itself. */
   options: ListrOptions
   /** Options for the current renderer of the task. */
@@ -76,25 +76,25 @@ export interface ListrTaskObject<Ctx, Renderer extends ListrRendererFactory> ext
   /** A hook to refresh render if desired. */
   renderHook$: Subject<void>
   /** Returns whether this task has subtasks. */
-  hasSubtasks: () => boolean
+  hasSubtasks(): boolean
   /** Returns whether this task is in progress. */
-  isPending: () => boolean
+  isPending(): boolean
   /** Returns whether this task is skipped. */
-  isSkipped: () => boolean
+  isSkipped(): boolean
   /** Returns whether this task has been completed. */
-  isCompleted: () => boolean
+  isCompleted(): boolean
   /** Returns whether this task has an active rollback task going on. */
-  isRollingBack: () => boolean
+  isRollingBack(): boolean
   /** Returns whether the rollback action was successful. */
-  hasRolledBack: () => boolean
+  hasRolledBack(): boolean
   /** Returns whether enabled function resolves to true. */
-  isEnabled: () => boolean
+  isEnabled(): boolean
   /** Returns whether this task has a prompt inside. */
-  isPrompt: () => boolean
+  isPrompt(): boolean
   /** Returns whether this task has been failed. */
-  hasFailed: () => boolean
+  hasFailed(): boolean
   /** Returns whether this task actually has a title. */
-  hasTitle: () => boolean
+  hasTitle(): boolean
 }
 
 export interface ListrTask<Ctx = ListrContext, Renderer extends ListrRendererFactory = any> {

